include/ensmallen_bits/problems/ackley_function.hpp

```diff
@@ -94,6 +98,15 @@ class AckleyFunction
    * @param gradient The function gradient.
    */
   void Gradient(const arma::mat& coordinates, arma::mat& gradient);
+
+  //! Get the value used for numerical stability.
+  double Epsilon() const { return epsilon; }
+  //! Modify the value used for numerical stability.
+  double& Epsilon() { return epsilon; }
```

Comment:
since these "get" and "set" accessor functions for `epsilon` are simply "pass-through" functions (ie. they don't protect anything), why not just make `epsilon` a public variable?

@zoq @rcurtin Is there a specific style / guideline when to use  "get" and "set" accessor functions in ensmallen?

Reply:
I don't think there is one, just replicated the style from mlpack: https://github.com/mlpack/mlpack/wiki/DesignGuidelines#naming-conventions. I guess the idea was to have the possibility to validate the input and use a similair style for both:

```c++
optimizer.Size() = 0.8;
optimizer.Size(0.8);
```

Personally, I don't mind to change the style for this particular use-case.

Reply:
Right, this was sort of a design decision we made early on with high hopes for being able to validate things (generally by using the form `optimizer.Size(0.8)` or equivalent), but even that has some drawbacks and so we ended up in a situation that looks like this.  There may be some utility where one wants to provide read-only access to something (although, a `const` public variable would do the same...).  At this point I might shrug and say that someday it might be worth changing to something safer or something simpler, but it would be a huge refactoring endeavor.